### ISSUE TITLE
Graph Customization

### ISSUE THREAD

**Issue by jefflee-icesat2** [cmacmackin/ford#226]
(Python newbie here) - 

I have modified a local copy of graphs.py that allow for customization of the graphs. I'm only just becoming marginally dangerous in Python, so it works, but I know it is not done the "correct" way. 

I did attempt to isolate the modifications from the main graph.py, but again, I don't think I did this the right way.
 
These changes allow for graphs such as the attached. The copy of graphs.py also has recursion commented out so that only the immediate neighbors of the base node are shown. My Fortran codebase is fairly large and the fully-recursed graphs nearly blew up my browser.  

I am willing to share my modified source if someone would like to pick it up and recode it the "right" way. It's nowhere near good enough to create a fork at this point. 

BTW- thanks for this project; its gonna save me lots of time.

![screen shot 2018-03-06 at 8 35 14 am](https://user-images.githubusercontent.com/37111053/37035088-5efa05fa-2119-11e8-9dea-ab4a0315e4e5.png)

**Comment by haraldkl** [comment]
> The copy of graphs.py also has recursion commented out so that only the immediate neighbors of the base node are shown

You can limit the depth of the graph with the `graph_maxdepth` option. Alternatively, you could also use `graph_maxnodes`, which limits the total number of nodes to draw in the graph.

**Comment by jefflee-icesat2** [comment]
Yep. graph_maxdepth is a nice enhancement. I pulled ford via pip when I started getting into this, so the version I pulled didn't have that enhancement in it.
